src/ethereum_test_types/blob_types.py

```diff
@@ -1,19 +1,449 @@
+        """Return filename this blob would have as string (with .json extension)."""
+        amount_cell_proofs: int = cast(int, fork.get_blob_constant("AMOUNT_CELL_PROOFS"))
+        return "blob_" + str(seed) + "_cell_proofs_" + str(amount_cell_proofs) + ".json"
+
+    @staticmethod
+    def get_filepath(fork: Fork, seed: int):
+        """Return the Path to the blob that would be created with these parameters."""
+        # determine amount of cell proofs for this fork (0 or 128)
+        would_be_filename: str = Blob.get_filename(fork, seed)
+
+        # create cached blobs dir if necessary
+        if not CACHED_BLOBS_DIRECTORY.exists():
+            CACHED_BLOBS_DIRECTORY.mkdir(
+                parents=True, exist_ok=True
+            )  # create all necessary dirs on the way
+
+        # return path to blob
+        return CACHED_BLOBS_DIRECTORY / would_be_filename
+
+    @staticmethod
+    def from_fork(fork: Fork, seed: int = 0, timestamp: int = 0) -> "Blob":
+        """Construct Blob instances. Fork logic is encapsulated within nested functions."""
+
+        def generate_blob_data(rng_seed: int = 0) -> Bytes:
+            """Calculate blob data deterministically via provided seed."""
+            # create local (independent) RNG object seeded with rng_seed
+            rng = random.Random(seed)
+
+            # generate blob
+            ints: list[int] = [
+                rng.randrange(cast(int, fork.get_blob_constant("BLS_MODULUS")))
+                for _ in range(cast(int, fork.get_blob_constant("FIELD_ELEMENTS_PER_BLOB")))
+            ]
+
+            encoded: list[bytes] = [
+                i.to_bytes(
+                    cast(int, fork.get_blob_constant("BYTES_PER_FIELD_ELEMENT")),
+                    cast(Literal["big"], fork.get_blob_constant("KZG_ENDIANNESS")),
+                )
+                for i in ints
+            ]
+            blob: bytes = b"".join(encoded)  # without 0x
+
+            return Bytes(blob)
+
+        def get_versioned_hash(commitment: Bytes, version: int = 1) -> Hash:
+            """Calculate versioned hash for a given blob."""
+            return Hash(bytes([version]) + sha256(commitment).digest()[1:])
+
+        def get_commitment(data: Bytes) -> Bytes:
+            """
+            Take a blob and returns a cryptographic commitment to it.
+
+            Note: Each cell holds the exact same copy of this commitment.
+            """
+            # sanity check
+            field_elements: int = cast(int, fork.get_blob_constant("FIELD_ELEMENTS_PER_BLOB"))
+            bytes_per_field: int = cast(int, fork.get_blob_constant("BYTES_PER_FIELD_ELEMENT"))
+            assert len(data) == field_elements * bytes_per_field, (
+                f"Expected blob of length "
+                f"{field_elements * bytes_per_field} but got blob of length {len(data)}"
+            )
+
+            # calculate commitment
+            commitment = ckzg.blob_to_kzg_commitment(data, Blob._trusted_setup)
```

Comment:
```suggestion
            commitment = ckzg.blob_to_kzg_commitment(data, Blob.trusted_setup())
```
Always use the `Blob.trusted_setup()` outside of that function, and that function is the only one that gets to access `Blob._trusted_setup` directly, because otherwise we have the risk of this code-path being reached first and the variable not being initialized.

Reply:
I don't think this could have happened but I agree that we should apply your suggestion anyway, it would never double load so I agree that this is safer at close to 0 performance cost

Reply:
Yet to be fixed in the latest commit.